### PR TITLE
tests: rgw_multisite playbook test refactor

### PR DESCRIPTION
**Pull request by guits**
Currently we create an object from the primary sites but we try to read
that object still from the master which doesn't make sense, we should
try to read it from a secondary site.

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

File: tests/functional/rgw_multisite.yml [hunk 1]
```diff
@@ -2,51 +2,56 @@
...
       register: result
       until: result is succeeded
-      when: not containerized_deployment | default(false) | bool
+      when: not containerized_deployment | bool
 
     - name: generate and upload a random 10Mb file - containerized deployment
-      command: >
-        podman run --rm --name=rgw_multisite_test --entrypoint=/bin/bash {{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }} -c 'dd if=/dev/urandom of=/tmp/testinfra.img bs=1M count=10; {{ s3cmd_cmd }} mb s3://testinfra; {{ s3cmd_cmd }} put /tmp/testinfra.img s3://testinfra'
+      shell: |
+        podman run --rm --name=rgw_multisite_test --entrypoint=dd {{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }} if=/dev/urandom of=/tmp/testinfra-{{ item.host }}-{{ item.instance_name }}.img bs=1M count=10;
```
Comment (guits) [inline tests/functional/rgw_multisite.yml:25]: this won't work

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Review by dsavineau — APPROVED** [review]

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by dsavineau** [comment]
jenkins test centos-container-rgw_multisite

**Review by dsavineau — COMMENTED** [review]
I'm wondering if we should keep the delay/retries on the get task from the secondary in case the data isn't replicated yet (rgw multisite is async AFAIK) or if we should executed the rgw_multisite.yml on the primary site just after the main playbook completion (instead of after the secondary site)
What's your thoughts on this ?

**Comment by guits** [comment]
> or if we should executed the rgw_multisite.yml on the primary site just after the main playbook 

According to the last test I've made, that wouldn't work

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-non_container-rgw_multisite

**Comment by guits** [comment]
jenkins test centos-container-rgw_multisite

**Review by dsavineau — APPROVED** [review]

**Pull request merged**